app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt

```diff
@@ -191,14 +197,24 @@ class HomeFragmentPresenter @Inject constructor(
         promotedStoryList.clear()
         if (it.recentStoryCount != 0) {
           it.recentStoryList.take(limit).forEach { promotedStory ->
-            val recentStory = PromotedStoryViewModel(activity, internalProfileId, storyEntityType)
+            val recentStory = PromotedStoryViewModel(
+              activity,
+              internalProfileId,
+              storyEntityType,
+              intentFactoryShimInterface
+            )
             recentStory.setPromotedStory(promotedStory)
             promotedStoryList.add(recentStory)
           }
         } else {
           // TODO(#936): Optimise this as part of recommended stories.
           it.olderStoryList.take(limit).forEach { promotedStory ->
-            val oldStory = PromotedStoryViewModel(activity, internalProfileId, storyEntityType)
+            val oldStory = PromotedStoryViewModel(
+              activity,
+              internalProfileId,
+              storyEntityType,
+              intentFactoryShimInterface
```

Comment:
Cut 'interface'. Ditto elsewhere for these shim variable names.

Reply:
Ok!